broker/value_added/value_added.py

```diff
@@ -0,0 +1,211 @@
+import numpy as np
+from warnings import warn as _warn
+from astropy.coordinates import SkyCoord
+from astroquery.irsa_dust import IrsaDust
+
+from . import xmatch as xm
+from . import classify as classify
+
+
+def get_value_added(alert_list, survey='ZTF'):
+    """ Calculates (or looks up) all value added products for each alert in alert_list.
+
+    Args:
+        alert_list (list): list of alert dicts
+        survey      (str): name of survey generating the alerts
+
+    Returns:
+        Lists of dictionaries of value added products.
+    """
+
+    # Cross Matches
+    xmatch_dicts_list = xm.get_xmatches(alert_list, survey=survey) # list of dictionaries
+
+    # Classification
+    # RAPID
+    light_curves, cx_data = format_for_rapid(alert_list, xmatch_dicts_list, survey=survey)
+    if len(light_curves) > 0:
+        class_dicts_list = classify.rapid(light_curves, cx_data, plot=False, use_redshift=True)
+    else:
+        print('\nThere are no alerts with enough information for classification.\n')
+        class_dicts_list = []
+
+    return xmatch_dicts_list, class_dicts_list
+
+
+def format_for_rapid(alert_list, xmatch_list, survey='ZTF'):
+    """ Creates a list of tuples formatted for RAPID classifier.
+
+    Args:
+        alert_list  (list): list of alert dicts
+        xmatch_list (list): list of cross matched objects as given by xm.get_xmatches()
+        survey       (str): name of survey generating the alerts
+
+    Returns:
+        light_curves (list): list of tuples formatted as required for input to RAPID.
+
+        cx_data      (list): list of dicts of candidate and xmatch info as needed for
+                             upload to BigQuery classification table.
+
+    """
+    # throw an error if received non-ZTF data
+    assert (survey=='ZTF'), "\nvalue_added.format_for_rapid() requires survey=='ZTF'"
+
+    light_curves = []
+    cx_dicts = {} # collect candidate and xmatch data to merge with RAPID results
+    oid_map = map_objectId_list(xmatch_list) # maps objectId's to list positions
+
+    for alert in alert_list:
+        oid = alert['objectId']
```

Comment:
This code block should probably be it's own function so we can handle schema changes.